enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,212 @@
+When the `--apply-manifests` option is specified a user can also optionally specify `--overwrite` which would cause the apply to bahave as the 'oc apply --overwrite' does currently. If the manifest exists on the cluster it will be updated.
+If the manifest exists on the cluster and `--overwrite` is not specified a warning will be displayed that the manifest could not be created.
+If `oc` encountered an error applying a manifest, it could optionally attempt to apply additional manifests before exiting non-zero, but would not be required to attempt any additional manifests.
+
+### Mirroring to a central registry in a fully air gapped environment
+
+A fully air gapped environment is one in which your cluster nodes cannot access the internet.
+For this reason you must mirror the images and manifests to a filesystem disconnected from that environment and then bring that host or removable media across that gap.
+Our documnetation refers to this as disconnected mirroring.
+It may also be the case that there are multiple clusters within the air gapped network.
+In such a case it makes sense to configure a central registry in the air gapped environment from which every cluster can pull upgrade imagery.
+Assuming the aforementioned use case and the use of removable media that will be sneakernetted across the gap, the steps for mirroring are:
+
+1. Connect the removable media to a system connected to the internet.
+2. Mirror the images and manifests to a directory on the removable media:
+	```console
+	$ oc adm release mirror --to-dir=<removable-media-path>/mirror quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+	```
+3. The enhanced `oc adm release mirror ...` command will output the path, similar to the example below, to a configmap created containing the image signature. Note this path as it will be used later when the configmap is applied to a cluster to be upgraded. 
+	```console
+	Configmap <removable-media-path>/mirror/manifests/sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92 created
+	...
+4. Run any required checks/scrubs on the removable media.
+5. Connect the removable media to a host within the air gapped environment that has access to both the central registry and to any cluster requiring upgrade.
+6. Upload the mirrored images to the central registry:
+	```console
+	$ oc image mirror --from-dir=<removable-media-path>/mirror file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+	```
+7. Use oc to login to a given cluster to be upgraded.
+8. Apply the the mirrored configmap, noted above, to the connected cluster for use by CVO for upgrade image verification:
+	```console
+	$ oc apply -f <removable-media-path>/mirror/manifests/sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+	```
+9. Perform the upgrade:
+	```console
+	$ oc adm upgrade --to-image REGISTRY/REPOSITORY/release@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+	```
```

Comment:
The above workflow looks good to me, but @smarterclayton  would be the right person to take final view.

Reply:
/assign @smarterclayton

Reply:
You'll need `--allow-explicit-upgrade` here (cf. [this comment][1]), unless you have separately pointed your cluster at a Cincinnati upstream which includes the edge you're trying to take here.

Also, I'd have expected `/release` to be part of `REPOSITORY`, but haven't actually run an `oc image mirror ...` to check.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1782982#c27